From 35a873bf3da694876a1390eb4ea324cfb3d64327 Mon Sep 17 00:00:00 2001
From: Gabriel Becker <ggasparb@redhat.com>
Date: Wed, 18 Jun 2025 14:04:34 +0200
Subject: [PATCH] Remove uefi/non-uefi from grub2 rules in case they do not
 need.

Products that have the same grub2 path for both UEFI/non-UEFI do not
need to set the platform and the products have now consolidated the use
of the grub2 rules to only select the ones that come from the non-UEFI
set of rules.
---
 linux_os/guide/system/bootloader-grub2/non-uefi/group.yml | 2 ++
 linux_os/guide/system/bootloader-grub2/uefi/group.yml     | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/linux_os/guide/system/bootloader-grub2/non-uefi/group.yml b/linux_os/guide/system/bootloader-grub2/non-uefi/group.yml
index b093bdad864..2a79674b363 100644
--- a/linux_os/guide/system/bootloader-grub2/non-uefi/group.yml
+++ b/linux_os/guide/system/bootloader-grub2/non-uefi/group.yml
@@ -5,4 +5,6 @@ title: 'Non-UEFI GRUB2 bootloader configuration'
 description: |-
     Non-UEFI GRUB2 bootloader configuration
 
+{{%- if grub2_boot_path != grub2_uefi_boot_path -%}}
 platform: non-uefi
+{{%- endif -%}}
diff --git a/linux_os/guide/system/bootloader-grub2/uefi/group.yml b/linux_os/guide/system/bootloader-grub2/uefi/group.yml
index e08747fe8c9..08f2e4ad9d0 100644
--- a/linux_os/guide/system/bootloader-grub2/uefi/group.yml
+++ b/linux_os/guide/system/bootloader-grub2/uefi/group.yml
@@ -5,7 +5,9 @@ title: 'UEFI GRUB2 bootloader configuration'
 description: |-
     UEFI GRUB2 bootloader configuration
 
+{{%- if grub2_boot_path != grub2_uefi_boot_path -%}}
 platform: uefi
+{{%- endif -%}}
 
 warnings:
     - functionality: |-
